lib/web/resources.go

```diff
@@ -50,27 +50,26 @@ func (h *Handler) checkAccessToRegisteredResource(w http.ResponseWriter, r *http
 		return nil, trace.Wrap(err)
 	}
 
-	resourceKinds := []string{types.KindNode, types.KindDatabaseServer, types.KindAppServer, types.KindKubeServer, types.KindWindowsDesktop}
-	for _, kind := range resourceKinds {
-		res, err := clt.ListResources(r.Context(), proto.ListResourcesRequest{
-			ResourceType: kind,
-			Limit:        1,
-		})
-		if err != nil {
-			// Access denied error is returned when user does not have permissions
-			// to read/list a resource kind which can be ignored as this function is not
-			// about checking if user has the right perms.
-			if trace.IsAccessDenied(err) {
-				continue
-			}
-			return nil, trace.Wrap(err)
-		}
+	res, err := clt.ListUnifiedResources(r.Context(), &proto.ListUnifiedResourcesRequest{
+		Kinds: []string{
+			types.KindNode,
+			types.KindDatabaseServer,
+			types.KindKubeServer,
+			types.KindWindowsDesktop,
+			types.KindAppOrSAMLIdPServiceProvider,
+		},
+		Limit:              1,
+		IncludeRequestable: true,
+		SortBy: types.SortBy{
+			Field:  services.SortByName,
+			IsDesc: false,
+		},
+	})
```

Comment:
tbh, I am still unclear on the intended behavior here:
1. Show the onboarding if the user has permission to create a new resource and there is exactly `zero` resource in cluster. OR 
2. Show the onboarding  if the user has permission to create a new resource but we could not fetch any resource accessible by that user. (could be due to `zero` resource in cluster or due to missing permission).

I checked this with a proper editor role that can create any type of new resource, but without the wildcard labels against a cluster that has resource configured with labels, the modal still pops up. Meaning, for users who has permission to create any type of resource but lacks access to existing resource gated behind specific labels, we will still show them the onboarding modal (the 2nd case)

Is that the behavior we want here?

Reply:
I believe so? # 2 makes sense to me, but I don't have much more ctx about this modal besides the fact it should also take SAML apps into account.

Reply:
Not a huge deal either way, I just wanted to confirm if we are not changing originally intended behavior here.  

Looking at the [original ticket](https://github.com/gravitational/teleport/issues/13997) for which this was implemented. Under **Auto Show Wizard** section, it says to show when
- User has the right perms
- Querying for each agent has no result

I think we wanted to implement Option 1? 

@kimlisa Can you please confirm the behavior, when do we want to show the onboarding modal?

Reply:
the way it currently works now was probably intended, I listed it as a requirement here: https://github.com/gravitational/webapps/pull/1121, which tells me that we were aware of it

Reply:
Thanks for confirming. 

It's option 2 then. In that case, can we utilize existing `clusterUnifiedResourcesGet` method here which already checks for `KindSAMLIdPServiceProvider` resource and handles all the other edge cases?

(I personally am in the option 1 bandwagon)

Reply:
Option 1 sounds great to me as well